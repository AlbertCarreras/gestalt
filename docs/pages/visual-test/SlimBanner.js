// @flow strict
import { type Node } from 'react';
import { SlimBanner, Flex, Box } from 'gestalt';

export default function Snapshot(): Node {
  return (
    <Box padding={1}>
      <Flex direction="column" gap={1} width={400}>
        <SlimBanner message="Your total audience includes all users who have seen your Pins." />
        {['error', 'success', 'warning', 'info'].map((status) => (
          <SlimBanner
            key={status}
            status={status}
            message={`This is a/an ${status} message.`}
            iconAccessibilityLabel="test"
            helperLink={{
              text: 'Learn more',
              accessibilityLabel: 'Learn more Pinterest.com',
              href: 'http://www.pinterest.com',
              onClick: () => {},
            }}
          />
        ))}
        {['error', 'success', 'warning', 'info'].map((status) => (
          <SlimBanner
            key={status}
            type="light"
            status={status}
            message={`This is a/an ${status} message.`}
            iconAccessibilityLabel="test"
          />
        ))}
      </Flex>
    </Box>
  );
}
