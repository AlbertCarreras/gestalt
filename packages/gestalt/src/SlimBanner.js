// @flow strict
import { type Node } from 'react';
import Box from './Box.js';
import Icon from './Icon.js';
import Link from './Link.js';
import Text from './Text.js';
import Flex from './Flex.js';
import { MESSAGING_TYPE_ATTRIBUTES } from './messaging.js';

type Props = {|
  /**
   * Helper [Link](https://gestalt.pinterest.systems/link) to be placed after the message. See the [helperLink variant](https://gestalt.pinterest.systems/slimbanner#helperLink) to learn more.
   */
  helperLink?: {|
    text: string,
    accessibilityLabel: string,
    href: string,
    onClick: ({|
      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
      dangerouslyDisableOnNavigation: () => void,
    |}) => void,
  |},

  /**
   * Label to describe the status iconâ€™s purpose. See the [Accessibility guidelines](https://gestalt.pinterest.systems/slimbanner#Accessibility) for details on proper usage.
   */
  iconAccessibilityLabel?: string,
  /**
   * Main content of SlimBanner. Content should be [localized](https://gestalt.pinterest.systems/slimbanner#Localization).
   *
   */
  message: string,
  /**
   * The type of status to display. See the [status variant](https://gestalt.pinterest.systems/slimbanner#Variants) to learn more.
   */
  status?: 'neutral' | 'error' | 'info' | 'warning' | 'success',
  /**
   * The type of SlimBanner. See the [type variant](https://gestalt.pinterest.systems/slimbanner#Variants) to learn more.
   */
  type?: 'default' | 'light',
|};

/**
 * [SlimBanner](https://gestalt.pinterest.systems/slimbanner) conveys brief information related to a specific section of a page. The message can relay success, warning, error or general information. Since they are about a specific section of a page or surface, SectionAlerts sit inside of a section, and not at the top of the page. For alerts that apply to the whole page, use [SlimBanner](https://gestalt.pinterest.systems/slimbanner).
 *
 * ![SlimBanner light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/cypress/integration/visual-test/__image_snapshots__/SlimBanner%20%230.png)
 * ![SlimBanner dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/cypress/integration/visual-test/__image_snapshots__/SlimBanner-dark%20%230.png)
 *
 */
export default function SlimBanner({
  helperLink,
  iconAccessibilityLabel,
  message,
  status = 'neutral',
  type = 'default',
}: Props): Node {
  const colorText = status === 'default' || type === 'light' ? 'default' : 'dark';

  return (
    <Box
      color={type === 'light' ? 'transparent' : MESSAGING_TYPE_ATTRIBUTES[status].backgroundColor}
      padding={type === 'light' ? 0 : 4}
      paddingY={type === 'light' ? 1 : 0}
      rounding={4}
      width="100%"
      minWidth={320}
    >
      <Flex gap={type === 'light' ? 2 : 4}>
        {status !== 'default' && iconAccessibilityLabel ? (
          <Icon
            accessibilityLabel={iconAccessibilityLabel}
            color={MESSAGING_TYPE_ATTRIBUTES[status].iconColor}
            icon={MESSAGING_TYPE_ATTRIBUTES[status].icon}
            size={16}
          />
        ) : null}
        <Box
          dangerouslySetInlineStyle={{ __style: status !== 'default' ? { marginTop: '-1px' } : {} }}
        >
          <Text inline color={colorText}>
            {message}{' '}
            {helperLink ? (
              <Text inline color={colorText}>
                <Link
                  accessibilityLabel={helperLink.accessibilityLabel}
                  href={helperLink.href}
                  onClick={helperLink.onClick}
                  target="blank"
                  inline
                >
                  {helperLink.text}
                </Link>
              </Text>
            ) : null}
          </Text>
        </Box>
      </Flex>
    </Box>
  );
}
