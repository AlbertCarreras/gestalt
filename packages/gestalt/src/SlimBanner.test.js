// @flow strict
import { create } from 'react-test-renderer';
import SlimBanner from './SlimBanner.js';

describe('SlimBanner', () => {
  it('renders default status with message', () => {
    const tree = create(<SlimBanner message="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders default status light type with message', () => {
    const tree = create(<SlimBanner type="light" message="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an icon with accessibility label', () => {
    const tree = create(
      <SlimBanner
        message="test"
        status="error"
        iconAccessibilityLabel="accessibility label with error"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an icon light type with accessibility label', () => {
    const tree = create(
      <SlimBanner
        message="test"
        status="error"
        type="light"
        iconAccessibilityLabel="accessibility label with error"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders helper link', () => {
    const tree = create(
      <SlimBanner
        message="test"
        helperLink={{
          text: 'Learn more',
          accessibilityLabel: 'Learn more Pinterest.com',
          href: 'http://www.pinterest.com',
          onClick: () => {},
        }}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
