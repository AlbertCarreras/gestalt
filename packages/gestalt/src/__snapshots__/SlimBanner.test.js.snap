// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlimBanner renders an icon light type with accessibility label 1`] = `
<div
  className="box paddingX0 paddingY0 rounding4"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap1 xsDirectionRow"
  >
    <div
      className="FlexItem"
    >
      <svg
        aria-hidden={null}
        aria-label="accessibility label with error"
        className="icon errorIcon iconBlock"
        height={16}
        role="img"
        viewBox="0 0 24 24"
        width={16}
      >
        <path
          d="test-file-stub"
        />
      </svg>
    </div>
    <div
      className="FlexItem"
    >
      <div
        className="box"
        style={
          Object {
            "marginTop": "-1px",
          }
        }
      >
        <span
          className="Text fontSize300 darkText alignStart breakWord fontWeightNormal"
        >
          test
           
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`SlimBanner renders an icon with accessibility label 1`] = `
<div
  className="box errorWeak paddingX4 paddingY4 rounding4"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap2 xsDirectionRow"
  >
    <div
      className="FlexItem"
    >
      <svg
        aria-hidden={null}
        aria-label="accessibility label with error"
        className="icon errorIcon iconBlock"
        height={16}
        role="img"
        viewBox="0 0 24 24"
        width={16}
      >
        <path
          d="test-file-stub"
        />
      </svg>
    </div>
    <div
      className="FlexItem"
    >
      <div
        className="box"
        style={
          Object {
            "marginTop": "-1px",
          }
        }
      >
        <span
          className="Text fontSize300 darkText alignStart breakWord fontWeightNormal"
        >
          test
           
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`SlimBanner renders default status light type with message 1`] = `
<div
  className="box paddingX0 paddingY0 rounding4"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap1 xsDirectionRow"
  >
    <div
      className="FlexItem"
    >
      <div
        className="box"
      >
        <span
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          test
           
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`SlimBanner renders default status with message 1`] = `
<div
  className="box paddingX4 paddingY4 rounding4 secondary"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap2 xsDirectionRow"
  >
    <div
      className="FlexItem"
    >
      <div
        className="box"
      >
        <span
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          test
           
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`SlimBanner renders helper link 1`] = `
<div
  className="box paddingX4 paddingY4 rounding4 secondary"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap2 xsDirectionRow"
  >
    <div
      className="FlexItem"
    >
      <div
        className="box"
      >
        <span
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          test
           
          <span
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            <a
              aria-label="Learn more Pinterest.com"
              className="link hideOutline tapTransition rounding0 inlineBlock underline hoverNoUnderline accessibilityOutline"
              href="http://www.pinterest.com"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn more
            </a>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
